pkg/vex/vex.go

```diff
@@ -169,12 +169,39 @@ func (vexDoc *VEX) ToJSON(w io.Writer) error {
 	return nil
 }
 
+// EffectiveStatement returns the latest VEX statement for a given product, that
+// is the statement that contains the latest data about impact to a given product.
+func (vexDoc *VEX) EffectiveStatement(product, vulnID string) (s *Statement) {
+	for i := range vexDoc.Statements {
+		if vexDoc.Statements[i].Vulnerability == vulnID {
+			isProduct := false
+			for _, p := range vexDoc.Statements[i].Products {
+				if p == product {
+					isProduct = true
+					break
+				}
+			}
+
+			if !isProduct {
+				continue
+			}
+
+			if s == nil {
+				s = &vexDoc.Statements[i]
+			} else if s.Timestamp.Before(*vexDoc.Statements[i].Timestamp) {
+				s = &vexDoc.Statements[i]
+			}
+		}
+	}
+	return s
+}
+
 // StatementFromID returns a statement for a given vulnerability if there is one.
 func (vexDoc *VEX) StatementFromID(id string) *Statement {
-	for _, statement := range vexDoc.Statements { //nolint:gocritic // turning off for rule rangeValCopy
-		if statement.Vulnerability == id {
-			logrus.Infof("VEX doc contains statement for CVE %s", id)
-			return &statement
+	logrus.Warn("vex.StatementFromID is deprecated and will be removed in an upcoming version")
```

Comment:
In addition to a logger warning, let's also follow this Go recommendation for handling deprecated items in a library:

(from https://go.dev/blog/godoc)

>Sometimes a struct field, function, type, or even a whole package becomes redundant or unnecessary, but must be kept for compatibility with existing programs. To signal that an identifier should not be used, add a paragraph to its doc comment that begins with “Deprecated:” followed by some information about the deprecation.

Reply:
done